### PR TITLE
[fix][test]: fix flaky test of ManagedCursorMetricsTest.testManagedCursorMetrics (#9919)

### LINKED ISSUE #9919: Flaky-test: ManagedCursorMetricsTest.testManagedCursorMetrics
Flaky-test: org.apache.pulsar.broker.stats.ManagedCursorMetricsTest.testManagedCursorMetrics
Number of failures: 10

org.apache.pulsar.broker.stats.ManagedCursorMetricsTest is flaky. The testManagedCursorMetrics test method fails sporadically.

```
java.lang.AssertionError: expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertTrue(Assert.java:45)
	at org.testng.Assert.assertTrue(Assert.java:55)
	at org.apache.pulsar.broker.stats.ManagedCursorMetricsTest.testManagedCursorMetrics(ManagedCursorMetricsTest.java:70)
```

Usage tip: To enable automatic navigation to failure message, open the following links with CTRL/CMD-click.
[example failure 2021-03-08T17:30:26.3999781Z](https://github.com/apache/pulsar/runs/2056548871?check_suite_focus=true#step:8:190)
[example failure 2021-03-08T11:28:16.6508831Z](https://github.com/apache/pulsar/runs/2055117809?check_suite_focus=true#step:8:1036)
[example failure 2021-03-08T11:14:45.8989014Z](https://github.com/apache/pulsar/runs/2055117809?check_suite_focus=true#step:8:638)
[example failure 2021-03-05T21:58:01.2183193Z](https://github.com/apache/pulsar/runs/2043229259?check_suite_focus=true#step:8:190)
[example failure 2021-03-05T09:34:45.4232866Z](https://github.com/apache/pulsar/runs/2037997236?check_suite_focus=true#step:8:190)
[example failure 2021-03-05T09:22:25.4442285Z](https://github.com/apache/pulsar/runs/2037974738?check_suite_focus=true#step:8:1014)
[example failure 2021-03-05T09:10:53.5286506Z](https://github.com/apache/pulsar/runs/2037974738?check_suite_focus=true#step:8:614)
[example failure 2021-03-05T08:59:35.6870954Z](https://github.com/apache/pulsar/runs/2037974738?check_suite_focus=true#step:8:198)
[example failure 2021-03-04T06:23:51.0098464Z](https://github.com/apache/pulsar/runs/2028449927?check_suite_focus=true#step:8:688)
[example failure 2021-03-04T05:17:18.0634430Z](https://github.com/apache/pulsar/runs/2028174394?check_suite_focus=true#step:8:286)

<details>
<summary>Full exception stacktrace</summary>
<code><pre>
java.lang.AssertionError: expected [true] but found [false]
	at org.testng.Assert.fail(Assert.java:99)
	at org.testng.Assert.failNotEquals(Assert.java:1037)
	at org.testng.Assert.assertTrue(Assert.java:45)
	at org.testng.Assert.assertTrue(Assert.java:55)
	at org.apache.pulsar.broker.stats.ManagedCursorMetricsTest.testManagedCursorMetrics(ManagedCursorMetricsTest.java:70)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

</pre></code>
</details>

### PR DESCRIPTION
**Pull request by wuxuanqicn**
Fixes #9919 

### Motivation

we need make sure broker executed all ack command and updated metrics, then we can generate and check metric

### Modifications

- enable AckReceipt
- await until ack procedure complete(ACK and ACK_RESPONSE command)

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

Need to update docs? 
    
- [x] `no-need-doc` 
  
  (Please explain why)

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java [hunk 1]
```diff
@@ -22,10 +22,11 @@
 import org.apache.bookkeeper.client.PulsarMockLedgerHandle;
 import org.apache.pulsar.broker.auth.MockedPulsarServiceBaseTest;
 import org.apache.pulsar.broker.stats.metrics.ManagedCursorMetrics;
-import org.apache.pulsar.client.api.Consumer;
-import org.apache.pulsar.client.api.Producer;
-import org.apache.pulsar.client.api.SubscriptionType;
+import org.apache.pulsar.client.api.*;
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java:25]: Please avoid using star import

**Review by codelipenghui — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java (hunk 1 above)
Comment (wuxuanqicn) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java]: done

**Review by RobertIndie — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java [hunk 2]
```diff
@@ -22,10 +22,17 @@
 import org.apache.bookkeeper.client.PulsarMockLedgerHandle;
 import org.apache.pulsar.broker.auth.MockedPulsarServiceBaseTest;
 import org.apache.pulsar.broker.stats.metrics.ManagedCursorMetrics;
-import org.apache.pulsar.client.api.Consumer;
-import org.apache.pulsar.client.api.Producer;
+
```
Comment (lhotari) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java:25]: The code style used in Pulsar code base doesn't have empty line between imports. you might want to configure IntelliJ to use the Pulsar code style and checkstyle configuration. the instructions are in the README, https://github.com/apache/pulsar#configure-code-style

**Comment by lhotari** [comment]
LGTM. Please rebase the changes so that this can be merged.

**Review by lhotari — APPROVED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java (hunk 2 above)
Comment (wuxuanqicn) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/ManagedCursorMetricsTest.java]: done

**Comment by lhotari** [comment]
closing and reopening to trigger a new build

**Pull request merged**